provider/azure/credentials.go

```diff
@@ -13,7 +15,11 @@ import (
 const (
 	credAttrAppId          = "application-id"
 	credAttrSubscriptionId = "subscription-id"
+	credAttrTenantId       = "tenant-id"
 	credAttrAppPassword    = "application-password"
+
+	clientCredentialsAuthType cloud.AuthType = "service-principal-secret"
+	deviceCodeAuthType        cloud.AuthType = "interactive"
```

Comment:
deviceCodeAuthType?  I don't understand how that relates to interactive.  Maybe a comment here would be appropriate.